### DISCUSSION TITLE
Faster deploy times with Copilot

### DISCUSSION THREAD

**Discussion by afgallo** [aws/copilot-cli#3480] [Q&A] [answered]
Is there a way to speed deployment time? In our experience, a single service with no addons or otherwise can take up to 7 min to deploy. We now have 3 services and a simple deployment can roughly 25 min.

I am wondering if anyone is dealing with the same issues and can shed some light.

<img width="826" alt="deploy time" src="https://user-images.githubusercontent.com/9999415/164153395-d853a796-2d04-45c9-83e4-0c83dcba5d06.png">

Thanks!

**Comment by Lou1415926** [comment, marked as answer]
Hello @afgallo ! Seems like the ECS service update is taking a while. A few things that could speed it up:
1. Decrease your contain health check `image.healthcheck`'s `interval`
2. Decrease your http health check `http.healthcheck`'s `interval`

These may help speed up the ECS service update!

We are also aiming to add [`recreate`](https://github.com/aws/copilot-cli/issues/3427) as a deployment option in our next release. You can use `"recreate"` to speed up the deployment in your test environment that can deal with some down time!

In addition, it may be also helpful to use [multi-stage build](https://docs.docker.com/develop/develop-images/multistage-build/) to speed up the image building process if that's taking a while as well!

**Reply by afgallo** [reply]
Thanks for replying, @Lou1415926 !

We already use multi-stage builds 😄 

Our interval was set to `180s`, will change to `5s` and that might do the trick!
